kong/plugins/mashape-analytics/buffer.lua

```diff
@@ -26,15 +32,27 @@ local ngx_timer_at = ngx.timer.at
 local table_insert = table.insert
 local table_concat = table.concat
 local table_remove = table.remove
+local string_sub = string.sub
 local string_len = string.len
 local string_rep = string.rep
 local string_format = string.format
+local math_pow = math.pow
+local math_min = math.min
 local setmetatable = setmetatable
 
 local MB = 1024 * 1024
-local MAX_BUFFER_SIZE = 1 * MB
+local MAX_BUFFER_SIZE = 500 * MB
```

Comment:
Hi @cb372 yes it's very large, but it isn't meant as a way to reject very large batches, it's mostly just a failsafe in case a client gets stuck and forgets to close a connection or something like that. The collector handles it just fine even with extremely large batches. If it reaches a size that big without being a bad socket, it's often due to huge bodies and we don't want to drop those.